lib/libc/newlib/CMakeLists.txt

```diff
@@ -73,6 +73,9 @@ if(CMAKE_C_COMPILER_ID STREQUAL "GNU")
 endif()
 zephyr_link_libraries(c)
 
+# newlib needs this to expose the full Zephyr C Library API
+zephyr_compile_definitions(_POSIX_C_SOURCE=200809L)
```

Comment:
I know the aim of this PR is not fixing newlib, but at least here https://github.com/zephyrproject-rtos/zephyr/pull/67052 it was not needed
Though I needed to add it in a few tests like for ex. in https://github.com/zephyrproject-rtos/zephyr/pull/67052/files#diff-afac009aa0a8f669dcbc7adb2e273455a0ba5dd4ac0fa03471801840bb987ccd
All places where I needed to add it seemed to appropriately require the macros due to the functions used.

Reply:
I believe this follows the intent of the Zephyr C library documentation -- Zephyr code should not require additional settings to gain access to any supported function, including those beyond the C standard library. For newlib, that requires exposing the entire POSIX API, at least until someone fixes newlib.

Reply:
No objection here. Hopefully we can come up with a way to do this on an as-needed basis, or at least make it specific to the platform or configured by the application.

Reply:
Darn - I guess this line might still be an issue because it changes cflags globally for the entire build, which would violate rules A.4 and likely A.5 of the [Coding Guidelines](https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html#rule-a-4-c-standard-library-usage-restrictions-in-zephyr-kernel).